### PR TITLE
[16.0]Adapt shopinvader invoice to fastapi

### PR DESCRIPTION
**Pull request by bguillot**

### CODE REVIEW

**Comment by sebastienbeau** [comment]
@lmignon

File: shopinvader_invoice/models/fastapi_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from fastapi import APIRouter
+
+from odoo import api, fields, models
+
+
+class FastapiEndpoint(models.Model):
+
+    _inherit = "fastapi.endpoint"
+
+    app: str = fields.Selection(
```
Comment (simahawk) [inline shopinvader_invoice/models/fastapi_endpoint.py:13]: why do you need this?

**Review by simahawk — COMMENTED** [review]
shouldn't this go to shopinvader_api_invoice?

File: shopinvader_invoice/schemas/invoice.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from odoo.api import Environment
+from odoo.addons.fastapi.depends import paging, authenticated_partner_env
+from odoo.addons.fastapi.schemas import PagedCollection, Paging
+
+
+class Amount(BaseModel, metaclass=ExtendableModelMeta):
+    amount_tax: float | None
+    amount_untaxed: float | None
+    amount_total: float | None
+    amount_due: : float = pydantic.Field(alias="amount_residual") | None
```
Comment (simahawk) [inline shopinvader_invoice/schemas/invoice.py:18]: this seems broken

File: shopinvader_invoice/services/invoice.py [hunk 3]
```diff
@@ -1,156 +1,191 @@
...
+async def search(
+    env: Environment = Depends(odoo_env),  # noqa: B008
+    paging: Paging = Depends(paging),
+    partner: ResPartner = Depends(authenticated_partner),
+) -> PagedCollection[AccountMove]:  # noqa: B008
+    """Get the list of partner invoices
+    """
+    domain = [("partner_id", "=", partner.id)]
+    count = env["account.move"].search_count(domain)
+    orders = env["account.move"].search(domain, limit=paging.limit, offset=paging.offset)
```
Comment (simahawk) [inline shopinvader_invoice/services/invoice.py:32]: *invoices

File: shopinvader_invoice/services/invoice.py [hunk 4]
```diff
@@ -1,156 +1,191 @@
...
+    dependencies=[Depends(authenticated_partner)],
+)
+async def search(
+    env: Environment = Depends(odoo_env),  # noqa: B008
+    paging: Paging = Depends(paging),
+    partner: ResPartner = Depends(authenticated_partner),
+) -> PagedCollection[AccountMove]:  # noqa: B008
+    """Get the list of partner invoices
+    """
+    domain = [("partner_id", "=", partner.id)]
```
Comment (simahawk) [inline shopinvader_invoice/services/invoice.py:30]:
```suggestion
    domain = [("partner_id", "=", partner.id), ("move_type", "=", "out_invoice"), ("state", "!=", "cancel")]
```
at least

File: shopinvader_invoice/services/invoice.py [hunk 5]
```diff
@@ -1,156 +1,191 @@
...
-            }
+@invoice_api_router.get(
+    "/invoices",
+    response_model=PagedCollection[AccountMove]
+    dependencies=[Depends(authenticated_partner)],
+)
+async def search(
+    env: Environment = Depends(odoo_env),  # noqa: B008
+    paging: Paging = Depends(paging),
+    partner: ResPartner = Depends(authenticated_partner),
```
Comment (simahawk) [inline shopinvader_invoice/services/invoice.py:26]: ResPartner not defined

File: shopinvader_invoice/services/invoice.py [hunk 6]
```diff
@@ -1,156 +1,191 @@
...
+) -> PagedCollection[AccountMove]:  # noqa: B008
+    """Get the list of partner invoices
+    """
+    domain = [("partner_id", "=", partner.id)]
+    count = env["account.move"].search_count(domain)
+    orders = env["account.move"].search(domain, limit=paging.limit, offset=paging.offset)
+    return PagedCollection[AccountMove](
+        total=count,
+        items=[AccountMove.from_orm(order) for invoice in invoices],
```
Comment (simahawk) [inline shopinvader_invoice/services/invoice.py:35]: here I get 
```
pydantic.errors.PydanticUserError: You must set the config attribute `from_attributes=True` to use from_orm
```

**Comment by simahawk** [comment]
@bguillot I hope you don't mind... I really need to move this fwd and this is really broken.
I refactored according latest practices here https://github.com/shopinvader/odoo-shopinvader/pull/1493

**Pull request closed without merging**